addons/web/static/tests/webclient/actions/load_state.test.js

```diff
@@ -0,0 +1,1708 @@
+
+describe.current.tags("headless");
+
+const actionRegistry = registry.category("actions");
+
+function logHistoryInteractions() {
+    patchWithCleanup(browser.history, {
+        pushState(state, _, url) {
+            expect.step(`pushState ${url}`);
+            return super.pushState(state, _, url);
+        },
+        replaceState(state, _, url) {
+            expect.step(`replaceState ${url}`);
+            return super.pushState(state, _, url);
+        },
+    });
+}
+
+/**
+ * @param {{ env: import("@web/env").OdooEnv }} [options]
+ */
+async function mountWebClient(options) {
+    await mountWithCleanup(WebClient, options);
+    // Wait for visual changes caused by a potential loadState
+    await animationFrame();
+    // wait for BlankComponent
+    await animationFrame();
+    // wait for the regular rendering
+    await animationFrame();
+}
+
+defineActions([
+    {
+        id: 1,
+        xml_id: "action_1",
+        name: "Partners Action 1",
+        res_model: "partner",
+        type: "ir.actions.act_window",
+        views: [[1, "kanban"]],
+    },
+    {
+        id: 2,
+        xml_id: "action_2",
+        type: "ir.actions.server",
+    },
+    {
+        id: 3,
+        xml_id: "action_3",
+        name: "Partners",
+        res_model: "partner",
+        mobile_view_mode: "kanban",
+        type: "ir.actions.act_window",
+        views: [
+            [false, "list"],
+            [1, "kanban"],
+            [false, "form"],
+        ],
+    },
+    {
+        id: 4,
+        xml_id: "action_4",
+        name: "Partners Action 4",
+        res_model: "partner",
+        type: "ir.actions.act_window",
+        views: [
+            [1, "kanban"],
+            [2, "list"],
+            [false, "form"],
+        ],
+    },
+    {
+        id: 27,
+        xml_id: "action_27",
+        name: "Partners Action 27",
+        res_model: "partner",
+        mobile_view_mode: "kanban",
+        type: "ir.actions.act_window",
+        path: "partners",
+        views: [
+            [false, "list"],
+            [1, "kanban"],
+            [false, "form"],
+        ],
+    },
+    {
+        id: 28,
+        xml_id: "action_28",
+        name: "Partners Action 28",
+        res_model: "partner",
+        type: "ir.actions.act_window",
+        views: [
+            [1, "kanban"],
+            [2, "list"],
+            [false, "form"],
+        ],
+    },
+    {
+        id: 1001,
+        tag: "__test__client__action__",
+        target: "main",
+        type: "ir.actions.client",
+        params: { description: "Id 1" },
+    },
+    {
+        id: 1002,
+        tag: "__test__client__action__",
+        target: "main",
+        type: "ir.actions.client",
+        params: { description: "Id 2" },
+    },
+    {
+        id: 1099,
+        xml_id: "wowl.client_action",
+        tag: "__test__client__action__",
+        target: "main",
+        type: "ir.actions.client",
+        params: { description: "xmlId" },
+    },
+]);
```

Comment:
do you need all these? For the other suites, if an action is only necessary in a test, I defined it in that test only (typically those actions are really specific).

Reply:
Yes, it's already a subset of [this list](https://github.com/odoo/odoo/blob/master/addons/web/static/tests/legacy/webclient/helpers.js#L151). Tests are duplicated in this suite with one that checks old url and other the new url.